wheatley/row_generation/complib_composition_generator.py

```diff
@@ -58,29 +59,37 @@ def removeprefix(string: str, prefix: str) -> str:
 class ComplibCompositionGenerator(RowGenerator):
     """ The RowGenerator subclass for generating rows from a CompLib composition. """
 
-    complib_url = "https://complib.org/composition/"
+    complib_url = "https://api.complib.org/composition/"
 
     def __init__(self, comp_id: int, access_key: Optional[str]=None) -> None:
+        # Generate URL and request the rows
         url = self.complib_url + str(comp_id) + "/rows"
         if access_key:
             url += "?accessKey=" + access_key
         request_rows = requests.get(url)
-
+        # Check for the status of the requests
         if request_rows.status_code == 404:
             raise InvalidCompError(comp_id)
-
         if request_rows.status_code == 403:
             raise PrivateCompError(comp_id)
-
         request_rows.raise_for_status()
+        # Parse the request responses as JSON
+        response_rows = json.loads(request_rows.text)
 
         # New line separated, skip the first line (rounds)
-        split_rows = request_rows.text.splitlines(False)[1::]
-        self.loaded_rows: List[Row] = [Row([Bell.from_str(bell) for bell in row]) for row in split_rows]
+        self.loaded_rows_and_calls: List[RowCall] = [
+            (
+                Row([Bell.from_str(bell) for bell in row]),
+                None if call == '' else call
```

Comment:
Looking at https://api.complib.org/composition/71929/rows
1) Do we need to trim the initial rounds / call of `Go method`?
2) Split the final call of `That's all; Stand`?

Looking at https://api.complib.org/composition/64722/rows
Changes of method appear as calls. Should the go into the chat?

Reply:
I think that it's good for method calls to appear as calls - they won't produce any sound but they are still calls so I think they should be treated the same.  Also, I (along with probably quite a lot of people) either have chat hidden or don't look at chat at all so I think many people would miss the method splices.  (I like your choice in example compositions, by the way! :wink:)

I'm not sure how to handle `That's All`/`Go (method)?`.

The `Stand` call should definitely be removed - this is a biproduct of the CompLib API being primarily designed for Handbell Stadium (and calling `Stand` after a touch makes no sense when ringing tower bell style).

I think all `RowGenerator`s should call `That's All` before the rounds at the end (my current model for implementing calls is to have the `RowGenerator`s always return all the necessary calls, and then have `Bot` decide whether or not to ignore them).

I feel like we need to have a way for a row generator to specify the starting method and I'm working on a way to allow a `RowGenerator` to specify calls that should be called before going into changes.  Calling `Go` should probably be implemented here too.

Reply:
> I think that it's good for method calls to appear as calls - they won't produce any sound but they are still calls so I think they should be treated the same. Also, I (along with probably quite a lot of people) either have chat hidden or don't look at chat at all so I think many people would miss the method splices.

Ok, I assume they just log console errors when it can't find the associated sound file?
While Chat does give a notification, so I'm sure people would look at it if needed I agree the centre is much better for putting calls. Also I think chat is hidden on smaller screens.

Reply:
I tend to split-screen RR and a video call, so I never have chat open when ringing.  I think we could also make a PR to RR along with this to make any call _starting with_ `Go` say 'Go next time' (and the same with `That's All`).  This way at least some audio feedback would be given when `Go <method>` or `That's All at Handstroke` is called.

Reply:
> `Go <method>` or `That's All at Handstroke`

To note our call handling in `Tower` also assumes a complete unique mapping of what's received (it will just issue a warning on other unknown calls), so I prefer not to complicate this unnecessarily. Can we just emit `Go`, `Go Plain Bob Minor` in quick succession?